lxd/warnings.go

```diff
@@ -380,3 +385,37 @@ func dbWarningToAPIWarning(d *Daemon, warning db.Warning) (api.Warning, error) {
 
 	return w, nil
 }
+
+func pruneResolvedWarningsTask(d *Daemon) (task.Func, task.Schedule) {
+	f := func(ctx context.Context) {
+		opRun := func(op *operations.Operation) error {
+			return pruneResolvedWarnings(ctx, d)
+		}
+
+		op, err := operations.OperationCreate(d.State(), "", operations.OperationClassTask, db.OperationWarningsPruneResolved, nil, nil, opRun, nil, nil)
+		if err != nil {
+			logger.Error("Failed to start prune resolved warnings operation", log.Ctx{"err": err})
+			return
+		}
+
+		logger.Info("Pruning resolved warnings")
+		_, err = op.Run()
+		if err != nil {
+			logger.Error("Failed to prune resolved warnings", log.Ctx{"err": err})
+		}
+		logger.Info("Done pruning resolved warnings")
+	}
+
+	return f, task.Daily()
+}
+
+func pruneResolvedWarnings(ctx context.Context, d *Daemon) error {
```

Comment:
@monstermunchkin this function doesn't seem correct given the spec says "Automatically prune resolved warnings after 24h", as if the task (for instance) runs daily at 23:59, then a task that has just been resolved at 23:58 would get removed, even though it has been resolved for <24 hours.

Reply:
No, you're right. The question now is, should we keep resolved warnings for 24h (and then remove them) or clean up resolved warnings every 24h. The latter would be easier, as we'd otherwise need to keep track of the time the warning was resolved.

Reply:
We should keep them for 24h, so probably run a task every minute and have a updated_at field in the DB so we can tell how long it's been in that state.

Reply:
Could we rather name it `resolved_at` as that's what we're really interested in?

Reply:
Actually, to be consistent, it should be `resolved_time` (or `updated_time`).